### PR TITLE
some more minor callback cleanup and development tweaks

### PR DESCRIPTION
**Pull request by ryanpetrello**

### CODE REVIEW

File: awx/conf/settings.py [hunk 1]
```diff
@@ -171,7 +171,6 @@ def __init__(self, cache, registry, encrypter=None, decrypter=None):
     def get(self, key, **kwargs):
         value = self.cache.get(key, **kwargs)
         value = self._handle_encryption(self.decrypter, key, value)
-        logger.debug('cache get(%r, %r) -> %r', key, empty, filter_sensitive(self.registry, key, value))
```
Comment (ryanpetrello) [inline awx/conf/settings.py:174]: This ends up getting called a *lot*, and _itself_ does *more* `settings.LOG_...` lookups.

File: awx/main/dispatch/worker/base.py [hunk 2]
```diff
@@ -148,7 +148,6 @@ def work_loop(self, queue, finished, idx, *args):
             finally:
                 if 'uuid' in body:
                     uuid = body['uuid']
-                    logger.debug('task {} is finished'.format(uuid))
```
Comment (ryanpetrello) [inline awx/main/dispatch/worker/base.py:151]: This is also really noisy, and kind of useless; the dispatcher has _other_ log messages it prints when a job actually succeeds/fails/cancels.

File: awx/main/utils/formatters.py [hunk 3]
```diff
@@ -139,37 +139,6 @@ def reformat_data_for_log(self, raw_data, kind=None):
             data = json.loads(data)
         data_for_log = {}
 
-        def index_by_name(alist):
```
Comment (ryanpetrello) [inline awx/main/utils/formatters.py:142]: I couldn't find any evidence that _this_ function was used anywhere.

File: awx/main/utils/formatters.py [hunk 4]
```diff
@@ -139,37 +139,6 @@ def reformat_data_for_log(self, raw_data, kind=None):
...
-                    name = subdict.get('name', None)
-                elif 'path' in subdict:
-                    name = subdict.get('path', None)
-                if name:
-                    # Logstash v2 can not accept '.' in a name
-                    name = name.replace('.', '_')
-                    adict[name] = subdict
-            return adict
-
-        def convert_to_type(t, val):
```
Comment (ryanpetrello) [inline awx/main/utils/formatters.py:158]: It's kind of silly to redefine this function on *every* log `.format()` call; I moved it below to the only block where it matters (if the performance logger is in, which I doubt many people outside of AWX developers are using).

File: awx/settings/development.py [hunk 5]
```diff
@@ -179,3 +179,4 @@
     os.environ['SDB_NOTIFY_HOST'] = os.popen('ip route').read().split(' ')[2]
 
 WEBSOCKET_ORIGIN_WHITELIST = ['https://localhost:8043', 'https://localhost:3000']
+AWX_CALLBACK_PROFILE = True
```
Comment (ryanpetrello) [inline awx/settings/development.py:182]: It's nice to just turn this on by default in the dev env so you can just `kill -SIGUSR1` on callback processes to profile them.

**Review by chrismeyersfsu — APPROVED** [review]

**Pull request merged**